### PR TITLE
[don't merge] ci: enable TEST_INTEGRATION_USE_SNAPSHOTTER

### PR DESCRIPTION
**Pull request by vvoland**
A draft PR to have the CI run integration tests with containerd integration enabled.

### CODE REVIEW

**Comment by thaJeztah** [comment]
@vvoland I did a quick rebase of this branch to get a fresh run of CI

**Comment by thaJeztah** [comment]
- I cherry-picked https://github.com/moby/moby/pull/45322 🍿 

/cc @laurazard @rumpl

**Comment by thaJeztah** [comment]
Seeing a lot of these failures; looks like something with restoring the frozen images? Haven't checked

```
TestDockerRegistryAuthHtpasswdSuite/TestLogoutWithExternalAuth
    docker_cli_logout_test.go:16: assertion failed: error is not nil: Error response from daemon: No such image: busybox:latest: [d5193e4374e5c] failed to download busybox
    --- FAIL: TestDockerRegistryAuthHtpasswdSuite/TestLogoutWithExternalAuth (0.67s)
```

File: hack/make/.integration-test-helpers [hunk 1]
```diff
@@ -96,7 +96,7 @@ run_test_integration_suites() {
 				--junitfile="${ABS_DEST}/${pkgname//./-}-junit-report.xml" \
 				--raw-command \
 				-- go tool test2json -p "${pkgname}" -t ./test.main ${pkgtestflags}
-		); then exit 1; fi
+		)
```
Comment (rumpl) [inline hack/make/.integration-test-helpers:99]: GHA is green now even if tests fail :/

**Comment by rumpl** [comment]
Woop woop, we can close this now!!!!

**Pull request closed without merging**